gcsfs/core.py

```diff
@@ -1534,7 +1534,8 @@ async def _get_file_request(
             callback.set_size(size)
 
             checker = get_consistency_checker(consistency)
-            os.makedirs(os.path.dirname(lpath), exist_ok=True)
+            lparent = os.path.dirname(lpath) or os.curdir
```

Comment:
Can we please add a test? Also, note that `LocalFileSystem()._parent(lpath)` does the right thing, but perhaps there's no real need to dogfood here.

Reply:
Thanks for the review. I have added the tests. 

Regarding `_parent()`, yeah, I don't see a real need here. Tbh, I try to avoid it as it calls `make_path_posix` which does a lot of things